src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -289,61 +276,45 @@ def execute(self, args):
 
         # TODO: This fundamentally alters the way Knack.invocation works here. Cannot be customized
         # with an event. Would need to be customized via inheritance.
+
+        expanded_args = list(_explode_list_args(parsed_args))
+        cmd = parsed_args.func
```

Comment:
Haven't looked too deeply since we talked, but from a first glance, it seems like this and `cmd` serve the same purpose. 
That being said, knack uses `func` and since we have testing that checks this, will look into this further at another time.